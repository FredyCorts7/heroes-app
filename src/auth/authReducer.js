import { types } from '../types/types';

export const authReducer = (state = {}, action) => {
  if (!action) return state;

  switch (action.type) {
    case types.LOGIN:
      return {
        ...action.payload,
        authenticated: true,
      };
    case types.LOGOUT:
      return {
        authenticated: false,
      };
    default:
      return state;
  }
};
