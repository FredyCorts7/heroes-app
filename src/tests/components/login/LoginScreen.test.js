import React from 'react';
import { mount } from 'enzyme';
import { AuthContext } from '../../../auth/AuthContext';
import { LoginScreen } from '../../../components/login/LoginScreen';
import '@testing-library/jest-dom';
import { types } from '../../../types/types';

describe('Testing to <LoginScreen />', () => {
  const authContext = {
    dispatch: jest.fn(),
  };

  const historyMock = {
    replace: jest.fn(),
  };

  const wrapper = mount(
    <AuthContext.Provider value={authContext}>
      <LoginScreen history={historyMock} />
    </AuthContext.Provider>
  );

  test('should to show succesfully', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should to be dispatch and navegation', () => {
    const handleLogin = wrapper.find('button').prop('onClick');
    handleLogin();

    expect(authContext.dispatch).toHaveBeenCalledWith({
      type: types.LOGIN,
      payload: {
        name: 'Camila',
      },
    });

    expect(historyMock.replace).toHaveBeenCalledWith('/');

    localStorage.setItem('lastPath', '/dc');
    handleLogin();

    expect(historyMock.replace).toHaveBeenCalledWith('/dc');
  });
});
