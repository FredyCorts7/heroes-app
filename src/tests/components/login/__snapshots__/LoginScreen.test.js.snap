// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing to <LoginScreen /> should to show succesfully 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-info"
    onClick={[Function]}
  >
    Log in
  </button>
</div>
`;
